graspologic/embed/case.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+from scipy.sparse.linalg import LinearOperator, eigsh
+from sklearn.preprocessing import normalize, scale
+
+from graspologic.utils import import_graph, to_laplacian, is_almost_symmetric
+from graspologic.embed.base import BaseSpectralEmbed
+
+
+class CovariateAssistedEmbed(BaseSpectralEmbed):
+    """
+    Perform Spectral Embedding on a graph with covariates, using the regularized graph Laplacian.
+
+    The Covariate-Assisted Spectral Embedding is a k-dimensional Euclidean representation
+    of a graph based on a function of its Laplacian and a vector of covariate features
+    for each node. For more information, see [1].
+
+    Parameters
+    ----------
+    alpha : float, optional, default = None
+        Tuning parameter to use:
+            -  None: Default to the ratio of the leading eigenvector of the Laplacian
+                     to the leading eigenvector of the covariate matrix.
+            - float: Use a particular alpha-value.
+
+    assortative : bool, default = True
```

Comment:
just curious, why do you feel like this is a fair default? In my head this is a stronger assumption but i havent' played with it much

Reply:
the paper says it's a good default value. Makes the eigenspace of L and XX^T contribute equally to the embedding

Reply:
i was talking about the assortative part

Reply:
oh, I picked it as a default because:

1. most networks that I've been exposed to have higher within-group connection probabilities than between-group
2. non-assortative is a bit higher computational cost, since we're calculating L(Lv) instead of Lv

not at all attached to assortative as a default though

Reply:
i thought assortative was LL do I have that backwards?

Reply:
Yes. The reason we do LL is to get rid of large negative eigenvalues. That only happens in the non-assortative case, since between-group probabilities are higher than within